driver/health/event_manager_default.go

```diff
@@ -0,0 +1,51 @@
+package health
+
+import (
+	"context"
+	"errors"
+	"reflect"
+)
+
+type DefaultHealthEventManager struct {
+	evtChan                chan interface{}
+	listenerEventTypeCache map[reflect.Type]Readiness
+}
+
+func NewDefaultHealthEventManager() *DefaultHealthEventManager {
+	return &DefaultHealthEventManager{
+		evtChan:                make(chan interface{}),
+		listenerEventTypeCache: make(map[reflect.Type]Readiness),
+	}
+}
+
+func (h *DefaultHealthEventManager) Dispatch(event interface{}) {
+	select {
+	case h.evtChan <- event:
+	default:
+	}
+}
+
+func (h *DefaultHealthEventManager) AddListener(listener Readiness) error {
```

Comment:
This whole function is meant to only be called on initialization, right? Might be better to make this a `New(listeners ...Readines) (*DefaultHealthEventManager, error)` function.

Reply:
I've done the changes but now I don't know how to handle the error return on the constructor since it's used in the function `HealthEventManager` from registry_memory.go that does not return an error (and should not I think).

What I have done for now is to log the error and return nil, which will probably make the code panic. However, as an error can only be generated with a wrong inputs on the constructor and since the inputs provided to the constructor is statically set in the code, that should never happen.

That being said, I'm not really happy with that, what's your thoughts ?